src/code_0807CC3C.c

```diff
@@ -0,0 +1,543 @@
+#include "global.h"
+#include "structures.h"
+#include "functions.h"
+#include "script.h"
+#include "entity.h"
+#include "save.h"
+
+typedef struct Thing {
+    u16 unk_1;
+    u16 unk_2;
+    u32 unk_3;
+} Thing;
+
+extern u8 gUnk_0811E514[];
+extern u8 gUnk_0811E510[];
+extern struct_0807D1C4 gUnk_0811E4BC[];
+extern const char gUnk_0811E4B4[8];
+extern ScriptExecutionContext gPlayerScriptExecutionContext;
+extern ScriptExecutionContext gScriptExecutionContextArray[0x20];
+
+extern void CreateSpeechBubbleExclamationMark(Entity*, u32, u32);
+extern void CreateSpeechBubbleQuestionMark(Entity*, u32, u32);
+extern void DeleteThisEntity(void);
+extern u32 sub_080B180C(u32, const char*);
+extern void sub_0801C4A0(u32);
+extern s32 __divsi3(s32, s32);
+extern void _call_via_r6(Entity*, ScriptExecutionContext*);
```

Comment:
Also a compiler intrinsic, for a function pointer that takes 6 arguments: foo(r0, r1, r2, r3, r4, r5)